src/workflows-app/SubmissionDetails.test.js

```diff
@@ -690,4 +782,349 @@ describe('Submission Details page', () => {
+    const getMethods = jest.fn(() => Promise.resolve(methodData));
+    const mockLeoResponse = jest.fn(() => Promise.resolve(mockAzureApps));
+    Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runs: {
+            get: getRuns,
+          },
+          runSets: {
+            get: getRunsSets,
+          },
+          methods: {
+            getById: getMethods,
+          },
+        },
+        Apps: {
+          listAppsV2: mockLeoResponse,
+        },
+        CromwellApp: {
+          workflows: () => {
+            return {
+              metadata: jest.fn(() => {
+                return Promise.resolve(runDetailsMetadata);
+              }),
+            };
+          },
+        },
+        AzureStorage: {
+          blobByUri: jest.fn(() => ({
+            getMetadataAndTextContent: () =>
+              Promise.resolve({
+                uri: 'https://someBlobFilePath.blob.core.windows.net/cromwell/user-inputs/inputFile.txt',
+                sasToken: '1234-this-is-a-mock-sas-token-5678',
+                fileName: 'inputFile.txt',
+                name: 'inputFile.txt',
+                lastModified: 'Mon, 22 May 2023 17:12:58 GMT',
+                size: '324',
+                contentType: 'text/plain',
+                textContent: 'this is the text of a mock file',
+                azureSasStorageUrl: 'https://someBlobFilePath.blob.core.windows.net/cromwell/user-inputs/inputFile.txt',
+              }),
+          })),
+          details: jest.fn(() => {
+            return Promise.resolve({ sas: { token: '1234-this-is-a-mock-sas-token-5678' } });
+          }),
+        },
+        Metrics: {
+          captureEvent,
+        },
+      };
+    });
+
+    // Act
+    await act(async () => {
+      render(
+        h(BaseSubmissionDetails, {
+          name: 'test-azure-ws-name',
+          namespace: 'test-azure-ws-namespace',
+          workspace: mockAzureWorkspace,
+          submissionId,
+          workflowId: '00001111-2222-3333-aaaa-bbbbccccdddd',
+          uri: 'https://coaexternalstorage.blob.core.windows.net/cromwell/user-inputs/inputFile.txt',
+        })
+      );
+    });
+
+    const executionLogsButtons = screen.getAllByRole('button', { name: 'Logs' });
+    expect(executionLogsButtons !== undefined);
+
+    for (const executionLogsButton of executionLogsButtons) {
+      // Act
+      await user.click(executionLogsButton);
+      // screen.logTestingPlaygroundURL();
+
+      // Assert
+      expect(screen.getByRole('dialog', { name: 'Workflow Execution Log' }) !== undefined);
+      expect(screen.getByText('File:') !== undefined);
+      expect(screen.getByText('workflow.log') !== undefined);
+      expect(screen.getByRole('link', { name: 'Download log' }) !== undefined);
+      expect(screen.getByRole('button', { name: 'Workflow Execution Log' }) !== undefined);
+      expect(screen.getByText('this is the text of a mock file') !== undefined);
```

Comment:
Same comment about `getBy...` queries and `expect`.

```suggestion
      screen.getByRole('dialog', { name: 'Workflow Execution Log' });
      screen.getByText('File:');
      screen.getByText('workflow.log';
      screen.getByRole('link', { name: 'Download log' });
      screen.getByRole('button', { name: 'Workflow Execution Log' });
      screen.getByText('this is the text of a mock file');
```

If you prefer the style of having `expect` in assertions, these could be something like:
```js
expect(element).toBeInTheDocument();
```
https://github.com/testing-library/jest-dom?tab=readme-ov-file#tobeinthedocument